src/test/java/reactor/netty/http/client/HttpClientTest.java

```diff
@@ -171,6 +171,31 @@ private DefaultFullHttpResponse response() {
 		return r;
 	}
 
+	/** This ensures that non-default values for the HTTP request line are visible for parsing. */
+	@Test
+	public void postVisibleToOnRequest() {
+		disposableServer = HttpServer.create().port(0)
+				.route(r -> r.post("/foo", (in, out) -> out.sendString(Flux.just("bar"))))
+				.bindNow();
+
+		final AtomicReference<HttpMethod> method = new AtomicReference<>();
+		final AtomicReference<String> path = new AtomicReference<>();
+
+		final HttpClientResponse response = createHttpClientForContextWithPort()
+				.doOnRequest((req, con) -> {
+					method.set(req.method());
+					path.set(req.path());
+				})
+				.post()
+				.uri("/foo")
+				.send(ByteBufFlux.fromString(Mono.just("foo")))
+				.response().block();
+
+		assertThat(response.status()).isEqualTo(HttpResponseStatus.OK);
+		assertThat(method.get()).isEqualTo(HttpMethod.POST);
+		assertThat(path.get()).isEqualTo("/foo");
```

Comment:
Remove `/` in front of `foo` because `req.path()` returns `the decoded path portion from the uri() without the leading and trailing '/' if present`

Reply:
gotcha. done!